subprojects/core/src/main/java/org/gradle/internal/classpath/transforms/InstrumentingClassTransform.java

```diff
@@ -309,13 +356,15 @@ private boolean visitINVOKESTATIC(String owner, String name, String descriptor)
         @Override
         public void visitInvokeDynamicInsn(String name, String descriptor, Handle bootstrapMethodHandle, Object... bootstrapMethodArguments) {
             if (isGroovyIndyCallsite(bootstrapMethodHandle)) {
+                // Handle for org.gradle.internal.classpath.Instrumented.bootstrap() method
                 Handle interceptor = new Handle(
                     H_INVOKESTATIC,
                     INSTRUMENTED_TYPE.getInternalName(),
-                    getBoostrapMethodName(interceptorFilter),
-                    GROOVY_INDY_INTERFACE_BOOTSTRAP_METHOD_DESCRIPTOR,
+                    "bootstrap",
```

Comment:
💭 A small simplification of this custom boostrap method, we pass filter name as a String so it can be read on the other side.